U/Uno/build_tarballs.jl

```diff
@@ -55,22 +56,25 @@ make -j${nproc}
 # Uno does not support `make install`. Manually copy for now.
 install -v -m 755 "uno_ampl${exeext}" -t "${bindir}"
 
-# Currently, Uno does not provide a shared library. THis may bbe useful in future once it has a C API.
-# ${CXX} -shared $(flagon -Wl,--whole-archive) libuno.a $(flagon -Wl,--no-whole-archive) -o "${libdir}/libuno.${dlext}" -L${libdir} -l${OMP} -l${LBT} -ldmumps -lmetis
+# Currently, Uno does not provide a shared library. This may be useful in the future once it has a C API.
+# We just check that we can generate it, but we don't include it in the tarballs.
+${CXX} -shared $(flagon -Wl,--whole-archive) libuno.a $(flagon -Wl,--no-whole-archive) -o libuno.${dlext} -L${libdir} -l${OMP} -l${LBT} -ldmumps -lmetis -lhsl
+# cp libuno.${dlext} "${libdir}/libuno.${dlext}
 """
 
 platforms = supported_platforms()
 filter!(p -> !(Sys.isfreebsd(p) && arch(p) == "aarch64"), platforms)
 platforms = expand_cxxstring_abis(platforms)
 
 products = [
-    # This LibraryProduct may be useful once Uno provides a C API. We omit it for nnow.
+    # This LibraryProduct may be useful once Uno provides a C API. We omit it for now.
     # LibraryProduct("libuno", :libuno),
     # We call this amplexe to match the convention of other JLL packages (like Ipopt_jll) that provide AMPL wrappers
     ExecutableProduct("uno_ampl", :amplexe),
 ]
 
 dependencies = [
+    Dependency(PackageSpec(name="HSL_jll", uuid="017b0a0e-03f4-516a-9b91-836bbd1904dd")),
```

Comment:
We really need compat bounds one stuff like this

Reply:
We can't do that on `HSL_jll.jl`. We don't have the official release number in the Julia registry.

Reply:
I don't know the right answer for this particular case. But I also mean it as a general comment for all JLL builds.

Reply:
The issue is how the upstream packages are versioned.
Sometimes they break things with a patch release, and in other cases, nothing is broken with a major one.

For the COIN-OR packages, everything is doomed by the developers, and the only solution is to use strict equality in the compat bounds to ensure that a new release of a dependency doesn't break everything. But it's not viable for all JLL, because we can't upgrade one package without recompiling all packages that depend on it if we want to use it.

Reply:
> But it's not viable for all JLL, because we can't upgrade one package without recompiling all packages that depend on it if we want to use it.

I kinda think this is what we should do